src/MotionControl.c

```diff
@@ -1557,30 +1567,26 @@ BOOL Ros_MotionControl_StartMotionMode(MOTION_MODE mode)
         Ros_Sleep(MOTION_START_CHECK_PERIOD);
     }
 
-updateStatus:
-    // Update status
     Ros_Controller_IoStatusUpdate();
 
     Ros_Debug_BroadcastMsg("%s: exit", __func__);
 
     //Required to allow motion api to work (Potential race condition)
     Ros_Sleep(200);
+    motion_readiness_code = Ros_Controller_GetNotReadySubcode(false);
```

Comment:
What's the purpose of the second call to `Ros_Controller_GetNotReadySubcode`?

Reply:
The first call ignored "tractable problems" which is what I'm calling `MotionNotReady` codes that can be resolved within the `Ros_MotionControl_StartMotionMode()` function. This second call makes sure that it actually meets all of the conditions for motion. It is logically the same as a `Ros_Controller_IsMotionReady()` call, like the calls before it, but I just used `Ros_Controller_GetNotReadySubcode` instead just in case it failed so that the proper `motion_readiness_code` can be returned

Reply:
Ok. that makes sense. Can you add some comment to that effect?